### PR TITLE
Add kb3x keyboard

### PR DESCRIPTION
**Pull request by DOIO2022**
Add kb3x keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/doio/kb3x/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    _LAY4,
+    _LAY5
+};
+
+// enum layer_keycodes { };
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_LAY0] = LAYOUT(
+		   TO(1),    RGB_MOD,    RGB_TOG,    RGB_HUI,    RGB_VAD,    RGB_VAI,
```
Comment (drashna) [inline keyboards/doio/kb3x/keymaps/default/keymap.c:34]: Could you replace all of the tabs with spaces?

File: keyboards/doio/kb3x/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    _LAY0,
+    _LAY1,
+    _LAY2,
+    _LAY3,
+    _LAY4,
+    _LAY5
+};
+
+// enum layer_keycodes { };
+
```
Comment (drashna) [inline keyboards/doio/kb3x/keymaps/default/keymap.c:30]:
```suggestion
```

File: keyboards/doio/kb3x/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    _LAY0,
+    _LAY1,
+    _LAY2,
+    _LAY3,
+    _LAY4,
+    _LAY5
+};
+
+// enum layer_keycodes { };
+
```
Comment (drashna) [inline keyboards/doio/kb3x/keymaps/via/keymap.c:30]:
```suggestion
```

File: keyboards/doio/kb3x/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_LAY0] = LAYOUT(
+		   TO(1),    RGB_MOD,    RGB_TOG,    RGB_HUI,    RGB_VAD,    RGB_VAI,
```
Comment (drashna) [inline keyboards/doio/kb3x/keymaps/via/keymap.c:34]: Tabs -> spaces

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/doio/kb3x/keymaps/default/keymap.c (hunk 1 above)
Comment (DOIO2022) [reply keyboards/doio/kb3x/keymaps/default/keymap.c]: @drashna  Hi , I've replaced it. Thank you!

File: keyboards/doio/kb3x/keymaps/via/keymap.c (hunk 4 above)
Comment (DOIO2022) [reply keyboards/doio/kb3x/keymaps/via/keymap.c]: I've replaced it. Thank you!

**Review by zvecr — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**